scripts/score_with_katago.py

```diff
@@ -0,0 +1,166 @@
+"""Rescores SGFs using KataGo's Tromp-Taylor scoring.
+
+KataGo scoring clears out opposite-color stones within pass-alive groups. For
+consistency throughout experiments, we should use KataGo scoring everywhere
+(including in experiments where otherwise KataGo isn't involved at all like
+baseline attacks vs. ELF).
+"""
+import argparse
+import getpass
+import os
+import shutil
+import subprocess
+from pathlib import Path
+
+import sgfmill.sgf
+
+
+def get_sgfs_in_file(sgf_file: Path):
+    """Get all SGFs in a file."""
+    if sgf_file.suffix == ".sgf":
+        # Assume entire file is one SGF.
+        with open(sgf_file) as f:
+            yield sgfmill.sgf.Sgf_game.from_string(f.read())
+    elif sgf_file.suffix == ".sgfs":
+        # Assume each line in the file is an SGF.
+        with open(sgf_file) as f:
+            for line in f:
+                yield sgfmill.sgf.Sgf_game.from_string(line)
+
+
+def get_sgfs_in_path(path: Path):
+    """Recursively get all SGFs in a path."""
+    yield from get_sgfs_in_file(path)
+    for dirpath, _, filenames in os.walk(path):
+        for f in filenames:
+            child_path = Path(os.path.join(dirpath, f))
+            yield from get_sgfs_in_file(child_path)
+
+
+def score_str_to_white_score(score_str: str) -> float:
+    """Convert a score string (e.g., B+35.5) to a numerical score for white."""
+    winner, score = score_str.split("+")
+    score = float(score)
+    if winner == "B":
+        score = -score
+    return score
+
+
+def get_white_score(game: sgfmill.sgf.Sgf_game) -> float:
+    """Get score for white from a game."""
+    score_str = game.get_root().get("RE")
+    return score_str_to_white_score(score_str)
+
+
+def main():
+    """Entrypoint for script."""
+    parser = argparse.ArgumentParser(
+        description="Rescores SGFs using KataGo's Tromp-Taylor scoring.",
+    )
+    parser.add_argument(
+        "sgf_path",
+        type=Path,
+        help="Path to directory (to be searched recursively) with input SGFs",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        type=Path,
+        help="Path to file to write output SGFs",
+        default=os.devnull,
+    )
+    args = parser.parse_args()
+    if args.output.is_file():
+        raise ValueError(f"Output file already exists: {args.output}")
+
+    tmp_dir = Path(f"/tmp/score-with-katago-{getpass.getuser()}")
```

Comment:
You could also use `tempfile.TemporaryDirectory` here, although idk if it's worth bumping everything up an indentation level